File: Array/003-RemoveElement/RemoveElement.js
var removeElement = function(nums, val) {
    let valIdx = 0

    for (let idx=0; idx < nums.length; idx++) {
        if (nums[idx] != val) {
            nums[valIdx] = nums[idx]
            valIdx += 1
        }
    }
    return valIdx
};

File: Array/003-RemoveElement/RemoveElement.go
func removeElement(nums []int, val int) int {
    valIdx := 0
    for i:=0; i < len(nums); i++ {
        if nums[i] != val {
            nums[valIdx] = nums[i]
            valIdx += 1
        }
    }
    return valIdx
}
